src/main/java/mate/academy/service/impl/AuthentificationServiceImpl.java

```diff
@@ -0,0 +1,53 @@
+package mate.academy.service.impl;
+
+import java.util.Optional;
+import mate.academy.dao.UserDao;
+import mate.academy.exception.AuthenticationException;
+import mate.academy.exception.RegistrationException;
+import mate.academy.lib.Inject;
+import mate.academy.lib.Service;
+import mate.academy.model.User;
+import mate.academy.service.AuthentificationService;
+import mate.academy.service.UserService;
+import mate.academy.util.HashUtil;
+
+@Service
+public class AuthentificationServiceImpl implements AuthentificationService {
+    @Inject
+    private UserDao userDao;
+    @Inject
+    private UserService userService;
+
+    @Override
+    public User login(String login, String password) throws AuthenticationException {
+        Optional<User> userFromDbOptional = userService.findByEmail(login);
+        if (userFromDbOptional.isEmpty()) {
+            throw new AuthenticationException("No user with such login: "
+                    + login);
+        }
+        User user = userFromDbOptional.get();
+        if (!(user.getPassword().equals(
+                (HashUtil.hashPassword(password, user.getSalt()))))) {
+            throw new AuthenticationException("Incorrect password for login:"
+                   + " " + login);
+        }
+        return user;
+    }
+
+    @Override
+    public User register(String login, String password) throws RegistrationException {
+        isLoginExists(login);
+        User user = new User();
+        user.setPassword(password);

```

Comment:
What if password is null?

Reply:
But why did you remove password validation?